src/openforms/js/components/admin/form_design/registrations/objectsapi/AddressNlObjectsApiVariableConfigurationEditor.js

```diff
@@ -0,0 +1,347 @@
+                  defaultMessage="House number addition Schema target"
+                  description="'House number addition Schema target' label"
+                />
+              }
+            >
+              <TargetPathSelect
+                id="houseNumberAdditionTargetPath"
+                name={`${namePrefix}.houseNumberAdditionTargetPath`}
+                index={index}
+                choices={choices}
+                mappedVariable={mappedVariable}
+              />
+            </Field>
+          </FormRow>
+          <FormRow>
+            <Field
+              name={`${namePrefix}.cityTargetPath`}
+              label={
+                <FormattedMessage
+                  defaultMessage="City Schema target"
+                  description="'City Schema target' label"
+                />
+              }
+              disabled={!deriveAddress}
+            >
+              <TargetPathSelect
+                id="cityTargetPath"
+                name={`${namePrefix}.cityTargetPath`}
+                index={index}
+                choices={choices}
+                mappedVariable={mappedVariable}
+                disabled={!deriveAddress}
+              />
+            </Field>
+          </FormRow>
+          <FormRow>
+            <Field
+              name={`${namePrefix}.streetNameTargetPath`}
+              label={
+                <FormattedMessage
+                  defaultMessage="Street name Schema target"
+                  description="'Street name Schema target' label"
+                />
+              }
+              disabled={!deriveAddress}
+            >
+              <TargetPathSelect
+                id="streetNameTargetPath"
+                name={`${namePrefix}.streetNameTargetPath`}
+                index={index}
+                choices={choices}
+                mappedVariable={mappedVariable}
+                disabled={!deriveAddress}
+              />
+            </Field>
+          </FormRow>
+        </Fieldset>
+      )}
+      <div style={{marginTop: '1em'}}>
+        <a href="#" onClick={e => e.preventDefault() || toggleJsonSchemaVisible()}>
+          <FormattedMessage
+            description="Objects API variable configuration editor JSON Schema visibility toggle"
+            defaultMessage="Toggle JSON Schema"
+          />
+        </a>
+        {jsonSchemaVisible && (
+          <pre style={{marginTop: '1em'}}>
+            {loading || (!mappedVariable.targetPath && !isSpecificTargetPaths) ? (
+              <FormattedMessage description="Not applicable" defaultMessage="N/A" />
+            ) : isSpecificTargetPaths ? (
+              JSON.stringify(ADDRESSNL_NESTED_PROPERTIES, null, 2)
+            ) : (
+              JSON.stringify(getTargetPath(mappedVariable.targetPath).jsonSchema, null, 2)
+            )}
+          </pre>
+        )}
+      </div>
+    </>
+  );
+};
+
+const TargetPathSelect = ({id, name, index, choices, mappedVariable, disabled}) => {
```

Comment:
Perhaps we can move this to a separate file. Seeing how AddressNlObjectsApiVariableConfigurationEditor and GenericObjectsApiVariableConfigurationEditor both use a simular component.

If it is indeed the same, it would be best to make 1 component that both AddressNL and Generic can use

Reply:
> both use a simular component

Similar != the same

This was deliberate to cut down on generic complexity - rather than have one component that must be able to support everything, it was opted to co-locate code and behaviour. We will for sure have more component types that require additional/flexible behaviours, and once a pattern emerges from that we can refactor/revisit.

Alex likes to mention the rule of three here - doing it once is okay, doing it a second time is also okay, once you're doing it for the the third time -> abstract it